lms/views/basic_lti_launch.py

```diff
@@ -103,10 +103,25 @@ def canvas_file_basic_lti_launch(self):
         Via. We have to re-do this file-ID-for-download-URL exchange on every
         single launch because Canvas's download URLs are temporary.
         """
-        self.context.js_config.add_canvas_file_id(
-            self.request.params["custom_canvas_course_id"],
-            self.request.params["file_id"],
+
+        course_id = self.request.params["custom_canvas_course_id"]
+        file_id = self.request.params["file_id"]
+
+        # Normally this would be done during `configure_module_item()` but
+        # Canvas skips that step. We are doing this to ensure that there is a
+        # module item configuration. As a result of this we can rely on this
+        # being around in future code.
+        self.assignment_service.set_document_url(
+            self.request.params["tool_consumer_instance_guid"],
+            self.request.params["resource_link_id"],
+            # This URL is mostly for show. We just want to ensure that a module
+            # configuration exists. If we're going to do that we might as well
+            # make sure this URL is meaningful.
+            document_url=f"canvas_file://course/{course_id}/file_id/{file_id}",
```

Comment:
For Blackboard files we went with `blackboard://content-resource/<ID>`.

The `content-resource` is because Blackboard doesn't really have a "files" feature as such, it has multiple ways to upload files that would need to be handled differently (e.g. they have different APIs for accessing them), and the one we're using is called "content resources". For Canvas there is only one file-uploading feature (as far as I know) and it's called "files".

So I think these ones should start with `canvas://file/` to be consistent with `blackboard://content-resource/`.

So I think that could suggest: `canvas://file/course/{course_id}/file/{file_id}`. Repeating `file` twice seems a bit awkward but the two `file`'s do play different roles.

Alternatively it's actually not too late to change the Blackboard ones to `blackboard_content_resource://...`

Reply:
No, I think that's better. I'll swap it over